plugins/jetpants_collins/topology.rb

```diff
@@ -274,12 +313,43 @@ def query_spare_assets(count, options={})
       # Now iterate in a single-threaded way for simplicity
       nodes.each do |node|
         db = node.to_db
-        if(db.usable_spare? && (!source || db.usable_with?(source)))
-          keep_nodes << node
-          break if keep_nodes.size >= count
+        if(db.usable_spare? &&
+          (
+            !source ||
+            (!source.pool && db.usable_with?(source)) ||
+            (
+              (!options[:for_pool] && source.pool && db.usable_in?(source.pool)) ||
+              (options[:for_pool] && db.usable_in?(options[:for_pool]))
```

Comment:
Yup, this method in general is a little heavy right now and is a target to be cleaned up a bit in future iterations.

Specifically if we're looking at ``source`` then it's basically an alias of ``options[:like]`` (which is a bit confusing stylistically) and is also optional.  The stuff around ``options[:for_pool]`` is more or less longhand for the ``compare_pool logic`` elsewhere in the code which was written later, but will eventually be unified to be consistent.